src/components/View-ArticleI-By-d/ArticleBox.jsx

```diff
@@ -17,7 +19,17 @@ export const ArticleBox = ({article}) => {
         )
     }
 
-   
+    const handleDownVote = (article_id) => {
+        setVote(vote-1)
+        patchArticle(article_id,-1)
+        .catch((err)=>setVote(vote+1))
```

Comment:
I can see what you're trying to do here, but state is currently mutating. If we click when offline, it will skip line 23 and jump straight to increasing the vote. What you can do to fix this is to pass a callback function into the state, use the current state as an argument and then increase/decrease this optimistically as you wish 🙂